packages/cli/src/commands/update.ts

```diff
@@ -0,0 +1,144 @@
+import { readFileSync } from 'fs';
+import { logger } from "../utils/logger.js";
+import { getUserPkgManager } from "../utils/getUserPkgManager.js";
+import { exec } from "child_process";
+import { installDependencies } from "../utils/installDependencies.js";
+import { resolvePath } from "../utils/parseNameAndPath.js";
+import { writeJSONFile } from "../utils/fileSystem.js";
+
+export async function updateCommand(path: string) {
+    try {
+        const resolvedPath = resolvePath(path);
+        const installedPackages = await getInstalledPackages(resolvedPath);
+        const triggerDevPackagesName = Object.keys(installedPackages.dependencies).filter(pkg => pkg.startsWith('@trigger.dev/'));
+
+        if (triggerDevPackagesName.length === 0) {
+            logger.info('No @trigger.dev packages found in package.json.');
+            return;
+        }
+
+        const triggerDevPackagesVersions = Object.fromEntries(triggerDevPackagesName.map(packageName => {
+            const currentVersion = installedPackages.dependencies[packageName].version;
+            return [packageName, currentVersion];
+        }));
+
+        const latestVersions = await getLatestVersions(triggerDevPackagesName, triggerDevPackagesVersions);
+        const updates: { [packageName: string]: string } = {};
+
+        triggerDevPackagesName.forEach(packageName => {
+            const currentVersion = installedPackages.dependencies[packageName].version;
+            const latestVersion = latestVersions[packageName];
+            if (latestVersion) {
+                if (semver.gt(latestVersion, currentVersion)) {
+                    updates[packageName] = latestVersion;
+                }
+            }
+        });
+
+
+        if (Object.keys(updates).length === 0) {
+            console.log('No updates available.');
+            return;
+        }
+
+        const confirm = await confirmUpdate(updates);
+
+        if (confirm) {
+            const updatedPackageJson = updatePackageJson(updates);
+            await writeJSONFile('package.json', updatedPackageJson);
+            logger.info('package.json updated.');
+
+            // Run 'npm install' to install the updated packages
+            logger.info('Running npm install...');
+            await installDependencies(resolvedPath);
+            logger.info('Dependencies updated.');
+        } else {
+            logger.info('Update canceled.');
+        }
+    } catch (error) {
+        logger.error('An error occurred:', error);
+    }
+}
+
+async function getInstalledPackages(resolvedPath: string): Promise<any> {
+    const manager = getUserPkgManager();
+    return new Promise((resolve, reject) => {
+        exec(`${manager} list --depth=0 --json`, {
+            cwd: resolvedPath
+        }, (error, stdout) => {
+            if (error) {
+                reject(error);
+            } else {
+                const installedPackages = JSON.parse(stdout.toString());
+                switch (manager) {
```

Comment:
This logic looks incorrect. Here's what I get when I run the list command using all three different package managers we support (pnpm, npm, and yarn) on a freshly created Next.js project

- `pnpm list --depth=0 --json` - https://jsonhero.io/j/Q9suKMhpFluT/editor
- `yarn list --depth=0 --json` - https://jsonhero.io/j/YG9GzRBVQ3Zj/editor
- `npm list --depth=0 --json` - https://jsonhero.io/j/FQEbMqFzH4xz/editor





```json